### PR TITLE
Add some queries for commonly-used third-part libraries

### PR DESCRIPTION
**Pull request by RicterZ**
Hi, I add some queries for commonly-used third-part libraries:

+ github.com/emicklei/go-restfu
+ github.com/jinzhu/gorm
+ github.com/jmoiron/sqlx
+ github.com/json-iterator/go

### CODE REVIEW

File: ql/src/semmle/go/frameworks/thirdpartlib/SQL.qll [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/**
```
[the remark this answers is not in the document]
Comment (max-schaefer) [reply ql/src/semmle/go/frameworks/thirdpartlib/SQL.qll:1]: Same suggestions regarding code style as with `GoRestfulSource` above.

File: ql/src/semmle/go/frameworks/thirdpartlib/HTTP.qll [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+module ThirdPartHttpLib {
+  /**
+   * Source from go-resultful
+   * Document: https://github.com/emicklei/go-restful
+   */
+  class GoRestfulSource extends DataFlow::Node, UntrustedFlowSource::Range {
+    GoRestfulSource() {
+      exists(
+	Method meth, string name |
+	meth.hasQualifiedName("github.com/emicklei/go-restful", "Request", name) and
```
[the remark this answers is not in the document]
Comment (intrigus-lgtm) [reply ql/src/semmle/go/frameworks/thirdpartlib/HTTP.qll:16]: Sorry for hijacking this.
Is there a "coding style" for set literals?
Is it ok to use them in functions like this `hasQualifiedName(_, ["foo", "bar", "foobar"])` (1)
or should one rather use something like this instead (2)
```codeql
exists(string name | name in ["foo", "bar", "foobar"] | hasQualifiedName(_, name))
```

If one already knows set literals (1) is probably understandable, but if I didn't know them I'd probably find that confusing.

[the remark this answers is not in the document]
Comment (max-schaefer) [reply ql/src/semmle/go/frameworks/thirdpartlib/HTTP.qll:16]: There isn't, both alternatives are fine.

**Comment by max-schaefer** [comment]
Superseded by https://github.com/github/codeql-go/pull/270.

**Pull request closed without merging**